src/neuroglancer/async_computation/decode_jpeg.ts

```diff
@@ -22,21 +22,26 @@ import {transposeArray2d} from 'neuroglancer/util/array';
 registerAsyncComputation(
     decodeJpeg,
     async function(
-        data: Uint8Array, width: number, height: number, numComponents: number,
+        data: Uint8Array, width: number|undefined, height: number|undefined, numComponents: number|undefined,
         convertToGrayscale: boolean) {
       let parser = new JpegDecoder();
       parser.parse(data);
       // Just check that the total number pixels matches the expected value.
-      if (parser.width * parser.height !== width * height) {
+      if (width && height && parser.width * parser.height !== width * height) {
```

Comment:
While it is unlikely to be a problem in this case, it would be better to explicitly check `width !== undefined && height !== undefined` --- just relying on the "truthy" conversion is error prone, since for example if the user specified a width of `0` that would not get checked, when instead it should lead to an error.